### ISSUE TITLE
Decide on a method and tolerance to make a selection fail

### ISSUE THREAD

**Issue by e-lo** [wsp-sag/network_wrangler#49]
**Question: What should the qualifications be for returning a "good selection"?**
This is showing a path from A-->B that goes along the one-way pair for the specified street because 6th St is one-way the other way.  This seems excessive. 

Right now this is controlled with the variable `RoadwayNetwork.MAX_SEARCH_BREADTH `
I have it set to 10, meaning that there could be a gap of 10 links between roads that have the name 6th (or a variation thereof).  We could:
1.  just lessen this, 
2. use a flag for `strict=True` to require that they all have the name (although many many links do not have a name) or 
3. require that they either have the name or the name is blank. This would then just be a problematic when the name is overlapping with something else. 
4. something I haven't thought of yet...

Thoughts?  cc: @DavidOry @josiekre @kulshresthaa 

![image](https://user-images.githubusercontent.com/794787/62971938-d3bc8600-bdc7-11e9-9c38-a90e4c40f224.png)

**Comment by josiekre** [comment]
This search method is specifically about letting users specific the links as "6th St" between node "A" and "B". Correct?

This seems like a user error and should stop in my opinion. If A -> B is one-way in the opposite direction than the user specified, the user is probably thinking about the project incorrectly. It should be changed from node B -> A by the user to get a correct result. Is that too stringent?

Sidenote: in the current implementation, can a user actually list the link IDs (e.g. OSMID, ShSt ID) rather than a name?

**Comment by e-lo** [comment]
> This seems like a user error and should stop in my opinion

Agreed (and this is a rather extreme example where part, but not all of the street is 2 way). Question is what are the cues for throwing the error.

**Comment by josiekre** [comment]
What does you mean by "overlapping with something else" in the second sentence in number 3?

> 3. require that they either have the name or the name is blank. This would then just be a problematic when the name is overlapping with something else.

**Comment by e-lo** [comment]
`Sidenote: in the current implementation, can a user actually list the link IDs (e.g. OSMID, ShSt ID) rather than a name?`
Right now we require a name, but that is a good call that if you are listing the IDs, we shouldn't require it.

Right now the algorithm should be fine with you listing IDs, but we should test.

**Comment by e-lo** [comment]
> "overlapping with something else"

i.e. if 6th Street  is sometimes called "Kermit the Frog Way", or if I-35 shares a facility with some other freeway but the naming is inconsistently one vs the other.

**Comment by josiekre** [comment]
Ah I see. I think 3 is the best we could do with names then, given the level of missing and overlapping names in Twin Cities (and a lot of other cities). 

I'm still thinking there may be too big of a problem when using "name" without some sort of user verification/visualization of the links that the changes are being applied to (regardless of any tolerance/criteria for failing). 

Ultimately the number of true fails depends on the messiness of the OSM data, but I could totally see a situation where a user thinks they are running a certain scenario with their model and in fact the model results are from a different one (changed lanes on a different road nearby and the information documenting this possible issue is hidden deep in a log file they forgot to check). It is very likely that at least 1 link on a long street will be mislabeled or labeled another valid name (not just missing). Although 1 out of x getting dropped might seem okay, it could be the most important link of the list of them (i.e. the link connecting to the freeway).

I'll test this out when working on Sprint 5 and see how I still feel about it.

**Comment by e-lo** [comment]
> It is very likely that at least 1 link on a long street will be mislabeled 

Agreed. That is why we are running an SP to link A to B for those situations (but we don't want to be too lenient so that we reject things like the above example)

> Visualization.

Totally agree! The TA version produced stacks of info confirming changes to each scenario including maps.  I think at the least we should have a project visual validation which goes a little beyond what we did in the current .ipynb.

**Comment by josiekre** [comment]
For anyone else who jumps in to read this...

SP = shortest path
TA = ?

**Comment by e-lo** [comment]
TA =  [SFCTA version](https://github.com/sfcta/networkwrangler)
SP = shortest path

( thx, I edited my post above to be more explicit )
